src/ops/base.rs

```diff
@@ -67,8 +249,46 @@ impl Ingredient for Base {
                 rs: Records,
                 _: &DomainNodes,
                 state: &StateMap)
-                -> Records {
-        rs.into_iter()
+                -> Option<Records> {
+        // Write incoming records to log before processing them if we are a durable node.
+        let records_to_return;
+        let mut copy_rs = rs.clone();
```

Comment:
Yeah, it isn't.  Wrote it like that because (1) I clearly haven't gotten a good grasp of Rust's borrowing semantics yet, and (2) other permutations of this that I tried all showed cryptic compilation error messages :)  Removed excessive cloning here.